tdp/core/models/state_enum.py

```diff
@@ -2,23 +2,19 @@
 # SPDX-License-Identifier: Apache-2.0
 
 from enum import Enum
+from typing import Union
 
+_PLANNED_STATE = "Planned"
+_RUNNING_STATE = "Running"
+_PENDING_STATE = "Pending"
+_SUCCESS_STATE = "Success"
+_FAILURE_STATE = "Failure"
+_HELD_STATE = "Held"
 
-class StateEnum(str, Enum):
-    """State enum.
-
-    Attributes:
-        SUCCESS (str): Success state.
-        FAILURE (str): Failure state.
-        PENDING (str): Pending state.
-    """
-
-    SUCCESS = "Success"
-    FAILURE = "Failure"
-    PENDING = "Pending"
 
+class _StateEnum(str, Enum):
     @classmethod
-    def has_value(cls, value):
+    def has_value(cls: Union["OperationStateEnum", "DeploymentStateEnum"], value: str):
```

Comment:
Why use a Union here ? You can not enumerate all `_StateEnum`'s children.
```suggestion
    def has_value(cls: _StateEnum, value: str):
```

Reply:
Fixed